week1/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func recoverError() {
+	if r := recover(); r != nil {
+		fmt.Println("recover from error: ", r)
+	}
+}
+
+func calFromStr(inputStr string) (result float64) {
+	inputArr := strings.Split(inputStr, " ")
+	for idx := 0; idx < len(inputArr)-1; idx += 2 {
+		if idx == 0 {
+			temp, err1 := strconv.ParseFloat(inputArr[idx], 64)
+			if err1 != nil {
+				errStr := fmt.Sprintf("parameter at [%v] must be numberic", idx)
+				panic(errors.New(errStr))
```

Comment:
In go, panic isn't used for bad input from users. The function should return error and let the caller handle that error